controller/lib/hal/hal_stm32_regs.h

```diff
@@ -316,69 +335,76 @@ struct ADC_Regs {
       REG smp17 : 3; // Sample time for channel 17
       REG smp18 : 3; // Sample time for channel 18
       REG rsvd2 : 5;
-    } samp;
+    } samp; // [RM] 16.6.6 ADC Sample Time Register 1 (ADC_SMPR1) (pg 464)
 
     REG rsvd1;
-    REG wdog[3]; // 0x20 - watchdog threshold registers
+    REG wdog[3]; // 0x20 - [RM] 16.6.8 ADC Watchdog
+                 // Threshold Register 1 (ADC_TR1) (pg 465)
     REG rsvd2;
 
     // 4x sequence registers.  These registers are used
     // to define the number of A/D readings and the
     // channel numbers being read.
     struct {
-      // sqr1
+      // [RM] 16.6.11 ADC Regular Sequence Register 1 (ADC_SQR1) (pg 468)
       REG len : 6;
       REG sq1 : 6;
       REG sq2 : 6;
       REG sq3 : 6;
       REG sq4 : 6;
       REG rsvd1 : 2;
 
-      // sqr2
+      // [RM] 16.6.12 ADC Regular Sequence Register 2 (ADC_SQR2) (pg 469)
       REG sq5 : 6;
       REG sq6 : 6;
       REG sq7 : 6;
       REG sq8 : 6;
       REG sq9 : 6;
       REG rsvd2 : 2;
 
-      // sqr4
+      // [RM] 16.6.13 ADC regular sequence register 3 (ADC_SQR3) (pg 470)
       REG sq10 : 6;
       REG sq11 : 6;
       REG sq12 : 6;
       REG sq13 : 6;
       REG sq14 : 6;
       REG rsvd3 : 2;
 
-      // sqr3
+      // [RM] 16.6.14 ADC Regular Sequence Register 4 (ADC_SQR4) (pg 471)
       REG sq15 : 6;
       REG sq16 : 6;
       REG rsvd4 : 20;
-    } seq;
+    } seq; // ADC Regular Sequence Register
 
-    REG data; // 0x40 - Regular data register
+    REG data; // 0x40 - Regular data register [RM] 16.6.15 (pg 471)
     REG rsvd3[2];
-    REG iSeq; // 0x4C - Injected sequence regiseter
+    REG iSeq; // 0x4C - Injected sequence regiseter [RM] 16.6.16 (pg 472)
     REG rsvd4[4];
-    REG offset[4]; // 0x60 - Offset registers
+    REG offset[4]; // 0x60 - Offset Register [RM] 16.6.17 (pg 473)
     REG rsvd5[4];
-    REG iData[4]; // 0x80 - Injected channel data
+    REG iData[4]; // 0x80 - - Injected Channel Data Register [RM] 16.6.18 (pg
```

Comment:
Can we make these comments fit on one line?  e.g. abbreviating "Injected chan data reg"?

Same below.